From: Rob Evers <revers@redhat.com>
Date: Mon, 4 Jan 2010 22:51:55 -0500
Subject: [scsi] lpfc: Fix vport register VPI after devloss timeout
Message-id: <20100104225202.24386.42503.sendpatchset@localhost.localdomain>
Patchwork-id: 22298
O-Subject: [RHEL5.5 PATCH 6/18] Fix vport fails to register VPI after devloss
	timeout. (CR: 96740)
Bugzilla: 549763
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>
RH-Acked-by: Mike Christie <mchristi@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=549763

Fix vport fails to register VPI after devloss timeout. (CR: 96740)

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/lpfc/lpfc_hbadisc.c b/drivers/scsi/lpfc/lpfc_hbadisc.c
index e42ddce..223a7f4 100644
--- a/drivers/scsi/lpfc/lpfc_hbadisc.c
+++ b/drivers/scsi/lpfc/lpfc_hbadisc.c
@@ -2392,6 +2392,7 @@ lpfc_mbx_cmpl_unreg_vpi(struct lpfc_hba *phba, LPFC_MBOXQ_t *pmb)
 		break;
 	}
 	vport->vpi_state &= ~LPFC_VPI_REGISTERED;
+	vport->fc_flag |= FC_VPORT_NEEDS_REG_VPI;
 	vport->unreg_vpi_cmpl = VPORT_OK;
 	mempool_free(pmb, phba->mbox_mem_pool);
 	/*
